GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -100,13 +100,47 @@ private abstract class Tree
             protected readonly Nodes Nodes;
             private readonly IGitUICommandsSource _uiCommandsSource;
 
+            protected readonly CancellationTokenSequence _reloadCancellation = new CancellationTokenSequence();
+
             protected Tree(TreeNode treeNode, IGitUICommandsSource uiCommands)
             {
                 Nodes = new Nodes(this);
                 _uiCommandsSource = uiCommands;
                 TreeViewNode = treeNode;
+
+                uiCommands.UICommandsChanged += (a, e) =>
+                {
+                    if (TreeViewNode?.TreeView != null)
+                    {
+                        TreeViewNode.TreeView.SelectedNode = null;
+                    }
+
+                    e.OldCommands.PostRepositoryChanged -= UICommands_PostRepositoryChanged;
+                    uiCommands.UICommands.PostRepositoryChanged += UICommands_PostRepositoryChanged;
+                };
+
+                uiCommands.UICommands.PostRepositoryChanged += UICommands_PostRepositoryChanged;
```

Comment:
This is the same pattern FormBrowse follows. The only leak is when the application is closed, but otherwise this makes sure to remove and add whenever the UICommands object has changed.